olcut-core/src/main/java/com/oracle/labs/mlrg/olcut/config/ConfigurationManager.java

```diff
@@ -647,6 +651,20 @@ private static boolean parseableAsBoolean(String input) {
     private static URL findURL(String input, String argumentName) {
         return AccessController.doPrivileged((PrivilegedAction<URL>)
                 () -> {
+                    int modIndex = input.indexOf(MODULE_SEPARATOR_CHAR);
+                    if (modIndex > 0) {
+                        try {
+                            String witnessClassName = input.substring(0, modIndex);
+                            String classPathName = input.substring(modIndex + 1);
+                            Class<?> witnessClass = Class.forName(witnessClassName);
+                            URL url = witnessClass.getResource(classPathName);
+                            if (url != null) {
+                                return url;
+                            } // else fall through to regular loading
+                        } catch (ClassNotFoundException e) {
+                            // fall through to regular loading
+                        }
+                    }
```

Comment:
If we had a vertical pipe but couldn't process the string, should we log a warning at that point rather than letting it fail lower down?  Or are we assuming the user knows what they're doing and they are just referencing a file that has a vertical pipe in the name?

Reply:
Logging a warning is probably sensible. That's a deeply tangled mess and it should probably log everything everywhere all the time because debugging that magic is hard.